ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,507 @@
+                "ipahealthcheck.ipa.host",
+                "--check",
+                "IPAHostKeytab",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        if result.returncode == 3:
+            assert data[0]["result"] == "ERROR"
+            assert data[0]["kw"]["msg"] == msg
+        else:
+            assert data[0]["result"] == "SUCCESS"
+        self.master.run_command(["systemctl", "start", dirsrv_service])
+
+    def test_source_ipahealthcheck_topology_IPATopologyDomainCheck(self):
+        """
+        Testcase checks default behaviour of check IPATopologyDomainCheck in
+        source ipahealthcheck.ipa.topology on IPA Master
+        """
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.topology",
+                "--check",
+                "IPATopologyDomainCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["suffix"] == "domain" or \
+                check["kw"]["suffix"] == "ca"
+
+    def test_source_ipa_roles_check_crlmanager(self):
+        """
+        This testcase checks the status of healthcheck tool
+        reflects correct information when crlgen is disabled
+        using ipa-crl-manage disable
+        """
+        self.master.run_command(["ipa-crlgen-manage", "disable"])
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.roles",
+                "--check",
+                "IPACRLManagerCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] == "True"
+
+    def test_ipa_healthcheck_remove(self):
+        """
+        This testcase checks the removal of of healthcheck tool
+        on replica and master
+        """
+        self.master.run_command(
+            ["dnf", "remove", "-y", "freeipa-healthcheck"])
+        self.replicas[0].run_command(
+            ["dnf", "remove", "-y", "freeipa-healthcheck"])
```

Comment:
The patch I'm going to propose will handle this in a generic way as well in tasks.